src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,30 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
 
+    private Map<String, UserData> cache = new ConcurrentHashMap<>();
+
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        try {
+            Thread.sleep(100);
```

Comment:
Using `Thread.sleep(100)` in an asynchronous method is not recommended as it blocks the thread. Consider using a non-blocking approach to simulate delay, such as `CompletableFuture.delayedExecutor` or similar techniques.